arangod/Replication2/Methods.cpp

```diff
@@ -320,7 +320,48 @@ struct ReplicatedLogMethodsCoordinator final
 
   auto insert(LogId id, TypedLogIterator<LogPayload>& iter) const
       -> futures::Future<std::pair<std::vector<LogIndex>, replicated_log::WaitForResult>> override {
-    THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+    auto path = basics::StringUtils::joinT("/", "_api/log", id, "multi-insert");
+
+    std::size_t payloadSize{0};
+    VPackBuilder builder{};
+    {
+      VPackArrayBuilder arrayBuilder{&builder};
+      while (const auto payload = iter.next()) {
+        if (payload.has_value()) {
+          builder.add(payload->slice());
+        } else {
+          builder.add(VPackSlice::emptyObjectSlice());
+        }
+        ++payloadSize;
```

Comment:
Here is a nice trick: `std::optional<T>` is convertible to `bool` and it converts to `true` if and only if `has_value()` returns true. That means that your `else` branch is never executed. The iterator signals that it has reached the end by returning a `std::nullopt`. Thus we can write
```
while (auto const payload = iter.next()) {
    builder.add(payload->slice()); // safe because otherwise we would have left the loop
    payloadSize++;
}
```